openedx/core/djangoapps/notifications/views.py

```diff
@@ -134,11 +181,17 @@ def patch(self, request, course_key_string):
             course_id=course_id,
             is_active=True,
         )
-        serializer = UserNotificationPreferenceSerializer(user_notification_preference, data=request.data)
-        if serializer.is_valid():
-            serializer.save()
-            return Response(serializer.data)
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+        form = NotificationPreferenceUpdateForm(request.data)
+        if not form.is_valid():
+            return Response(form.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        updated_notification_preference = update_notification_preference(
+            user_notification_preference.notification_preference_config, form.cleaned_data
+        )
+        user_notification_preference.notification_preference_config = updated_notification_preference
+        user_notification_preference.save()
+        serializer = UserNotificationPreferenceSerializer(user_notification_preference)
```

Comment:
These can be moved to update method of serializer and validation can be added there too, 
The code would look something like this 
```python
serializer = UserNotificationPreferenceSerializer(user_notification_preference)
if serializer.is_valid()
     serializer.save()
Response(serializer.data, status=status.HTTP_200_OK)

```